### PR TITLE
Remove sysroot-abi feature flag from proc-macro-test

### PR DESCRIPTION
**Pull request by Veykril**
Closes https://github.com/rust-lang/rust-analyzer/issues/16194

### CODE REVIEW

File: crates/proc-macro-srv/proc-macro-test/build.rs [hunk 1]
```diff
@@ -17,7 +17,6 @@ use cargo_metadata::Message;
 
 fn main() {
     println!("cargo:rerun-if-changed=imp");
-    println!("cargo:rerun-if-env-changed=PROC_MACRO_TEST_TOOLCHAIN");
```
Comment (lnicola) [inline crates/proc-macro-srv/proc-macro-test/build.rs:20]: Don't we want this?

Comment (Veykril) [reply crates/proc-macro-srv/proc-macro-test/build.rs:20]: do we? I checked our CI and bootstrap in rustc, neither of which use this.

**Comment by Veykril** [comment]
Hmm, this doesn't work as easily as I wanted to ...

File: crates/proc-macro-srv/proc-macro-test/build.rs (hunk 1 above)
Comment (lnicola) [reply crates/proc-macro-srv/proc-macro-test/build.rs:20]: The build script uses it, so it will catch whoever tries to change it (you?) off-guard.

Comment (Veykril) [reply crates/proc-macro-srv/proc-macro-test/build.rs:20]: Well no, I removed it from the build script altogether

Comment (lnicola) [reply crates/proc-macro-srv/proc-macro-test/build.rs:20]: Oh, sorry, I misread the diff.

**Pull request merged**